src/Component/Symfony/Request/Parameters.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Component\Resource\Symfony\Request;
+
+use Symfony\Component\HttpFoundation\ParameterBag;
+use Symfony\Component\HttpKernel\Kernel;
+
+if (Kernel::MAJOR_VERSION === 6) {
+    class Parameters extends ParameterBag
+    {
+        /**
+         * @param mixed $default
+         */
+        public function get(string $key, $default = null): mixed
+        {
+            $result = parent::get($key, $default);
+
+            if (null === $result && $default !== null && $this->has($key)) {
+                $result = $default;
+            }
+
+            return $result;
+        }
+    }
+} else {
+    class Parameters extends ParameterBag
+    {
+        /**
+         * @param mixed $default
+         *
+         * @return mixed
+         */
+        public function get(string $key, $default = null)
+        {
+            $result = parent::get($key, $default);
+
+            if (null === $result && $default !== null && $this->has($key)) {
+                $result = $default;
+            }
+
+            return $result;
+        }
+    }
+}
```

Comment:
We are lacking class alias definition below new classes like here: https://github.com/api-platform/core/blob/2.7/src/Symfony/Validator/EventListener/ValidationExceptionListener.php#L75

Reply:
Same is true for the all other files

Reply:
everything is on the src/deprecation.php
https://github.com/Sylius/SyliusResourceBundle/pull/564/files#diff-ad6b069705ee18edffce5eb4909fdd1014ef68c42fea8b336ea417e2ab25fe3b